src/scmdata/dataframe.py

```diff
@@ -1024,6 +1024,70 @@ def resample(self, rule: str = "AS", **kwargs: Any) -> ScmDataFrame:
         )
         return self.interpolate(list(target_dts), **kwargs)
 
+    def time_mean(self, style: str) -> ScmDataFrame:
+        """
+        Take time mean of self
+
+        TODO: decide whether this would be better as some sort of groupby operation...
+
+        TODO: decide whether to keep metadata or not
+
+        Parameters
+        ----------
+        style : ['year', 'year_beginning_of_year', 'year_end_of_year']
```

Comment:
Strings aren't bad. That's how numpy and pandas do it.

You could use `A`, `AS` and `AE` to line up with the resampling names

Reply:
There isn't an `AE` resampler, but it follows the pattern

Reply:
`A` is end of year so will make `AM` middle of year? https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#dateoffset-objects